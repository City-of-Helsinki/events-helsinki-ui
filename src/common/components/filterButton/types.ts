export type FilterType =
  | 'category'
  | 'date'
  | 'dateType'
  | 'division'
  | 'place'
  | 'publisher'
  | 'target'
  | 'text'
  | 'minAge'
  | 'maxAge'
  | 'exactAge';
