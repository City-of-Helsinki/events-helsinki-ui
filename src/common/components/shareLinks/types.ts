export interface ShareLinkProps {
  sharedLink: string;
}
