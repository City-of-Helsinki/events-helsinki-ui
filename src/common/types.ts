import { ReactElement } from "react";

export interface AutosuggestMenuItem {
  text: string;
  type: string;
  value: string;
}

export interface Category {
  icon?: ReactElement;
  text: string;
  value: string;
}
