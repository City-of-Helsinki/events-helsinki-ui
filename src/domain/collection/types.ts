import { CollectionDetails, LocalizedObject } from "../../generated/graphql";

export type CollectionInList = Pick<
  CollectionDetails,
  "id" | "curatedEvents" | "eventListQuery"
> & { title: Pick<LocalizedObject, "en" | "fi" | "sv"> };
