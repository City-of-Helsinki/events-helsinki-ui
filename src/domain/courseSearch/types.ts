export interface Filters {
  categories: string[];
  dateTypes: string[];
  divisions: string[];
  places: string[];
  text: string[];
  start: Date | null;
  end: Date | null;
  isFree?: boolean;
  alsoOngoingCourses?: boolean;
}

export interface MappedFilters {
  categories: string[];
  dateTypes?: string[];
  divisions: string[];
  isFree?: boolean;
  alsoOngoingCourses?: boolean;
  places: string[];
  publisher?: string | null;
  text: string[];
  start: string | null;
  end: string | null;
}
