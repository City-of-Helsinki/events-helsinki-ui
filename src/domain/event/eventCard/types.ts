import { ROUTES } from '../../app/constants';

export enum EventType {
  EVENT = 'event',
  COURSE = 'course',
}

export const EVENT_ROUTE_MAPPER = {
  [EventType.EVENT]: ROUTES.EVENT,
  [EventType.COURSE]: ROUTES.COURSE,
};

export const EVENTS_ROUTE_MAPPER = {
  [EventType.EVENT]: ROUTES.EVENTS,
  [EventType.COURSE]: ROUTES.COURSES,
};
