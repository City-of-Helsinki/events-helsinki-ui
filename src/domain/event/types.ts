import {
  Division,
  EventDetails,
  Image,
  Keyword,
  LocalizedObject,
  Maybe,
  Offer
} from "../../generated/graphql";

type OfferInList = Pick<Offer, "isFree"> & {
  description: Maybe<LocalizedObject>;
  price: Maybe<LocalizedObject>;
  infoUrl: Maybe<LocalizedObject>;
};

type KeywordInList = Pick<Keyword, "id"> & {
  name: Maybe<LocalizedObject>;
};

type LocationInList = {
  addressLocality: Maybe<LocalizedObject>;
  divisions: Maybe<
    Array<
      Pick<Division, "type"> & {
        name: Maybe<LocalizedObject>;
      }
    >
  >;
  name: Maybe<LocalizedObject>;
  streetAddress: Maybe<LocalizedObject>;
};

export type EventUiKeyword = {
  id: string;
  name: string;
};

export type EventInList = Pick<EventDetails, "id" | "startTime" | "endTime"> & {
  images: Array<Pick<Image, "id" | "name" | "url">>;
  keywords: Array<KeywordInList>;
  location: Maybe<LocationInList>;
  name: LocalizedObject;
  offers: Array<OfferInList>;
};
