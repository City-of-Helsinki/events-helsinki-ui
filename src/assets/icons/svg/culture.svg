
<svg width="33px" height="55px" viewBox="0 0 33 55" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <g transform="translate(-183.000000, -520.000000)">
    <g transform="translate(183.000000, 519.000000)">
      <path d="M18.722019,38.6656 C26.4513524,37.7232 32.4288762,31.792 32.4288762,24.6336 C32.4288762,20.6096 31.5238286,17.0384 30.6495429,13.5856 C29.802781,10.2336 29.0013524,7.0672 29.0013524,3.568 L29.0013524,1.44 L4.13601905,1.44 L4.13601905,3.568 C4.13601905,7.0672 3.33620952,10.2336 2.48944762,13.5856 C1.6151619,17.0384 0.710114286,20.6096 0.710114286,24.6336 C0.710114286,31.792 6.68601905,37.7232 14.4169714,38.6656 L14.4169714,51.096 L2.87801905,51.0944 L2.87801905,55.3504 L30.2609714,55.3504 L30.2609714,51.0944 L18.722019,51.096 L18.722019,38.6656 Z M8.36335238,5.6976 L24.7756381,5.6976 C25.0184952,8.872 25.7551619,11.784 26.470781,14.6192 C26.7881143,15.8704 27.0989714,17.0976 27.3628762,18.3232 C26.7249714,18.4992 26.1923048,18.7152 25.7163048,18.9104 C24.8954476,19.2464 24.298019,19.4928 23.2472571,19.4928 C22.1964952,19.4928 21.6023048,19.2464 20.7798286,18.9104 C19.7792571,18.5008 18.5358286,17.9904 16.5686857,17.9904 C14.6031619,17.9904 13.3581143,18.5008 12.3591619,18.9104 C11.5366857,19.2464 10.9408762,19.4928 9.89011429,19.4928 C8.83935238,19.4928 8.24354286,19.2464 7.42268571,18.9104 C6.94668571,18.7152 6.4124,18.4992 5.77449524,18.3232 C6.0384,17.0976 6.34925714,15.8704 6.66820952,14.6192 C7.38220952,11.784 8.11887619,8.872 8.36335238,5.6976 L8.36335238,5.6976 Z M5.01678095,24.6336 C5.01678095,23.9376 5.0524,23.2544 5.10906667,22.5808 C5.32278095,22.6592 5.53811429,22.7472 5.77125714,22.8416 C6.7556381,23.2448 7.98125714,23.7488 9.89011429,23.7488 C11.8005905,23.7488 13.0229714,23.2448 14.0089714,22.8416 C14.8508762,22.4976 15.4596381,22.2464 16.5686857,22.2464 C17.6777333,22.2464 18.2864952,22.4976 19.130019,22.8416 C20.1144,23.2448 21.3384,23.7488 23.2472571,23.7488 C25.1577333,23.7488 26.3817333,23.2448 27.3677333,22.8416 C27.5992571,22.7472 27.8162095,22.6592 28.0299238,22.5808 C28.0865905,23.2544 28.1205905,23.9376 28.1205905,24.6336 C28.1205905,30.0992 22.938019,34.544 16.5686857,34.544 C10.1993524,34.544 5.01678095,30.0992 5.01678095,24.6336 L5.01678095,24.6336 Z"></path>
    </g>
  </g>
</svg>
