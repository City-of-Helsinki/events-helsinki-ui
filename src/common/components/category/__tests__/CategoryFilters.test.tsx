import { shallow } from "enzyme";
import React from "react";

import { ReactComponent as CultureIcon } from "../../../../assets/icons/svg/culture.svg";
import { Category as CategoryType } from "../../../types";
import CategoryFilters from "../CategoryFilters";

it("CategoryFilters matched snapshot", () => {
  const el = shallow(
    <CategoryFilters
      categories={[
        {
          icon: <CultureIcon />,
          text: "test",
          value: "value"
        }
      ]}
      onClickCategory={(category: CategoryType) => {}}
    />
  );
  expect(el.html()).toMatchSnapshot();
});
