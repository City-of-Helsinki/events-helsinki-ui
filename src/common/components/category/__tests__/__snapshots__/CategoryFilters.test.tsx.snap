// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilters matched snapshot 1`] = `"<div class=\\"categoryFilters\\"><div class=\\"categoryFilter\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M13.3822163,16.4386198 C16.5500995,16.047777 19,13.587923 19,10.6191108 C19,8.95023225 18.6290644,7.46914399 18.2707366,6.03715992 C17.9236894,4.64698076 17.5952223,3.33377571 17.5952223,1.88254811 L17.5952223,1 L7.40411413,1 L7.40411413,1.88254811 C7.40411413,3.33377571 7.07631055,4.64698076 6.72926344,6.03715992 C6.37093563,7.46914399 6,8.95023225 6,10.6191108 C6,13.587923 8.44923689,16.047777 11.6177837,16.4386198 L11.6177837,21.5938952 L6.88852024,21.5932316 L6.88852024,23.3583278 L18.1114798,23.3583278 L18.1114798,21.5932316 L13.3822163,21.5938952 L13.3822163,16.4386198 Z M9.13669542,2.76575979 L15.8633046,2.76575979 C15.9628401,4.08228268 16.2647644,5.28998009 16.5580624,6.46582614 C16.6881221,6.98473789 16.8155275,7.49369608 16.9236894,8.00199071 C16.6622429,8.07498341 16.4439283,8.16456536 16.2488388,8.2455209 C15.9124088,8.3848706 15.6675514,8.48706038 15.2368945,8.48706038 C14.8062376,8.48706038 14.5627074,8.3848706 14.2256138,8.2455209 C13.8155275,8.07564698 13.3059058,7.86396815 12.4996682,7.86396815 C11.6940942,7.86396815 11.1838089,8.07564698 10.7743862,8.2455209 C10.4372926,8.3848706 10.1930989,8.48706038 9.76244194,8.48706038 C9.331785,8.48706038 9.08759124,8.3848706 8.75116125,8.2455209 C8.55607167,8.16456536 8.33709356,8.07498341 8.07564698,8.00199071 C8.18380889,7.49369608 8.31121433,6.98473789 8.44193762,6.46582614 C8.734572,5.28998009 9.03649635,4.08228268 9.13669542,2.76575979 L9.13669542,2.76575979 Z M7.76509622,10.6191108 C7.76509622,10.3304579 7.77969476,10.0471135 7.80291971,9.7677505 C7.89051095,9.80026543 7.97876576,9.83676178 8.07431984,9.87591241 C8.4777704,10.0431321 8.9800929,10.2521566 9.76244194,10.2521566 C10.5454545,10.2521566 11.0464499,10.0431321 11.450564,9.87591241 C11.7956204,9.73324486 12.0451228,9.62906437 12.4996682,9.62906437 C12.9542137,9.62906437 13.203716,9.73324486 13.549436,9.87591241 C13.9528865,10.0431321 14.4545455,10.2521566 15.2368945,10.2521566 C16.0199071,10.2521566 16.521566,10.0431321 16.9256802,9.87591241 C17.0205707,9.83676178 17.1094891,9.80026543 17.1970803,9.7677505 C17.2203052,10.0471135 17.2342402,10.3304579 17.2342402,10.6191108 C17.2342402,12.885866 15.1101526,14.7292634 12.4996682,14.7292634 C9.88918381,14.7292634 7.76509622,12.885866 7.76509622,10.6191108 L7.76509622,10.6191108 Z\\"></path></svg><span>test</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div></div>"`;
